presto-hive/src/main/java/com/facebook/presto/hive/HiveConnectorFactory.java

```diff
@@ -65,14 +66,14 @@
 {
     private final String name;
     private final ClassLoader classLoader;
-    private final ExtendedHiveMetastore metastore;
+    private final Optional<ExtendedHiveMetastore> metastore;
 
-    public HiveConnectorFactory(String name, ClassLoader classLoader, ExtendedHiveMetastore metastore)
+    public HiveConnectorFactory(String name, ClassLoader classLoader, @Nullable ExtendedHiveMetastore metastore)
```

Comment:
Should we make `HiveConnectorFactory` take `Optional` as well?

Reply:
good idea.